tests/filecheck/transforms/jax-use-donated-arguments.mlir

```diff
@@ -0,0 +1,55 @@
+// RUN: xdsl-opt %s -p jax-use-donated-arguments --split-input-file --verify-diagnostics | filecheck %s
+
+builtin.module {
+func.func public @main(%arg0: tensor<2x3xf32>, %arg1: tensor<3x4xf32>, %arg2: tensor<2x4xf32> {tf.aliasing_output = 0 : i32}) -> (tensor<2x4xf32>) {
+    %0 = tensor.empty() : tensor<2x4xf32>
+    %cst = arith.constant 0.000000e+00 : f32
+    %1 = linalg.fill ins(%cst : f32) outs(%0 : tensor<2x4xf32>) -> tensor<2x4xf32>
```

Comment:
We try to make the tests as targeted as possible. In this case, the fact that the results are produced by linalg is incidental, and any other operation should work just as well.
```suggestion
    %res = "test.op"() : () -> tensor<2x4xf32>
```
Similarly below